libs/shared/src/main/java/io/crate/common/collections/Lists.java

```diff
@@ -104,6 +106,21 @@ public static <T> List<T> concatUnique(List<? extends T> list1, Collection<? ext
         return result;
     }
 
+    public static Collection<?> flatUnique(Collection<?> list) {
+        if (!(list instanceof List<?>)) {
+            throw new IllegalArgumentException("Cannot flatten unless it is a nested array");
+        }
+        Set<Object> result = new HashSet<>();
```

Comment:
with HashSet we loose the ordering of the initial values (and potentially have some performance impact, if the most frequent values are provided first in the list).
Maybe use LinkedHashSet.